airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py

```diff
@@ -813,8 +819,8 @@ def test_read(
 
         assert records, "At least one record should be read using provided catalog"
 
-        if should_validate_schema:
-            self._validate_schema(records=records, configured_catalog=configured_catalog)
+        if should_validate_schema:  # TODO Conditional logic
```

Comment:
Yep - there was weird potential conditional logic here where I'd want to run the extra fields check even if `should_validate_schema` was false... without failing on other schema validation errors which would be weird. 

As far as I can see, only one connector makes use of `should_validate_schema: false`. For that reason I'd be in support of removing it and making it mandatory. But that also sort of goes against our idea that connectors can opt out of anything if they want to 🤔 but I think there have to be limitations to that

Reply:
> But that also sort of goes against our idea that connectors can opt out of anything if they want to 🤔

I'm not sure I'm very comfortable with this idea. I'd love to be sure a specific set of guarantees are provided by a 🟢 CAT. I think it's a bit out of the scope of this PR but it'd deserve an issue so that we can decide what a basic CAT runs.

Reply:
I'm more on your side for this question but I'm interested in hearing @evantahler's take on this specific question re: removing should_validate_schema

I agree that we should have a separate issue determining which tests should be canonically skippable

Reply:
Which connector opts out of the `should_validate_schema` test today?  I can't find it ([search](https://github.com/search?q=repo%3Aairbytehq%2Fairbyte+should_validate_schema&type=code)).

+1 that we can revisit the choice to make some tests opt-out-able or not in another PR.   I'm comforted that at the very least, GA connectors need high-strictness, and won't opt out of any checks. 

brainstorming: I could imagine some source that doesn't have structured records (mongo?) that might have trouble with `should_validate_schema=True`... but then again, it would `fail_on_extra_fields` too.

Reply:
Search should be just for `validate_schema` which is the input key. [Here](https://github.com/airbytehq/airbyte/blob/6a10ae3e05fc4499fc2d6aa86fb0279050d36350/airbyte-integrations/connectors/source-azure-table/acceptance-test-config.yml#L18) is the connector

Reply:
Trying to run it to see if it would succeed without that, but its failing locally anyway 🙃

Reply:
Iiinteresting. But yeah it seems like it would fail the new test too, hmm 

It seems like in the case that we say don't do schema validation, we probably want to also skip failing on extra fields. But the other way around is not necessarily the case - if we skip failing on extra fields bc the connector hasn't been updated to fix it yet, we still want to run schema validation like we have been

Reply:
To me that means keeping 2 separate flags but letting the user know that the value of fail_on_extra_fields will always resolve to false if validate_schema is false

Potentially out of scope of this PR - seeing as we were questioning why a connector would want to bypass schema validation, there should def be bypass_reasons for declaring that

Reply:
In this case I'm leaving in the note about eventually sunsetting fail_on_extra_fields - I think we only expect that sources that need to skip schema validation would expect to fail on extra fields, and we always run it within schema validation.